package/utils/px5g-wolfssl/px5g-wolfssl.c

```diff
@@ -0,0 +1,309 @@
+	to = INT_MAX;
+    strftime(tstr, sizeof(tstr), "%Y%m%d%H%M%S", gmtime(&to));
+
+    fprintf(stderr, "Generating selfsigned certificate with subject '%s'"
+		    " and validity %s-%s\n",
+	subject, fstr, tstr);
+
+    memset(pemBuf, 0, sizeof(pemBuf));
+    ret = wc_InitCert(&newCert);
+    if (ret != 0) {
+	fprintf(stderr, "Init Cert failed: %d\n", ret);
+	return ret;
+    }
+    newCert.isCA = 0;
+
+    if (type == EC_KEY_TYPE) {
+	ret = wc_MakeCert(&newCert, derBuf, sizeof(derBuf), NULL, &ecKey, rng);
+    } else {
+	ret = wc_MakeCert(&newCert, derBuf, sizeof(derBuf), &rsaKey, NULL, rng);
+    }
+    if (ret <= 0) {
+	fprintf(stderr, "Make Cert failed: %d\n", ret);
+	return ret;
+    }
+
+    if (type == EC_KEY_TYPE) {
+	newCert.sigType = CTC_SHA256wECDSA;
+	ret = wc_SignCert(newCert.bodySz, newCert.sigType, derBuf, sizeof(derBuf), NULL,
+	    &ecKey, rng);
+    } else {
+	newCert.sigType = CTC_SHA256wRSA;
+	ret = wc_SignCert(newCert.bodySz, newCert.sigType, derBuf, sizeof(derBuf), &rsaKey,
+	    NULL, rng);
+    }
+    if (ret <= 0) {
+	fprintf(stderr, "Sign Cert failed: %d\n", ret);
+	return ret;
+    }
+    derSz = ret;
+
+    ret = wc_DerToPem(derBuf, derSz, pemBuf, sizeof(pemBuf), CERT_TYPE);
+    if (ret <= 0) {
+	fprintf(stderr, "DER to PEM failed: %d\n", ret);
+	return ret;
+    }
+    pemSz = ret;
+
+    ret = write_file(pemBuf, pemSz, certpath);
+    if (ret != 0) {
+	fprintf(stderr, "Write Cert failed: %d\n", ret);
+	return ret;
+    }
+
+    if (type == EC_KEY_TYPE) {
+	wc_ecc_free(&ecKey);
+    } else {
+	wc_FreeRsaKey(&rsaKey);
+    }
+    return 0;
+}
+
+int dokey(WC_RNG* rng, int type, char** arg)
+{
+    ecc_key ecKey;
+    RsaKey rsaKey;
+    int ret;
+    int curve = ECC_SECP256R1;
+    int keySz = 2048;
+    int exp = 65537;
+    char* path = NULL;
+    bool pem = true;
+
+    while (*arg && **arg == '-') {
+	if (!strncmp(*arg, "-out", 4) && arg[1]) {
+	    path = arg[1];
+	    arg++;
+	} else if (!strncmp(*arg, "-3", 2)) {
+	    exp = 3;
+	} else if (!strncmp(*arg, "-der", 4)) {
+	    pem = false;
+	}
+	arg++;
+    }
+
+    if (*arg && type == RSA_KEY_TYPE) {
+	keySz = (unsigned int)atoi(*arg);
+    } else if (*arg) {
+	if (!strncmp(*arg, "P-256", 5)) {
+	    curve = ECC_SECP256R1;
+	} else if (!strncmp(*arg, "P-384", 5)) {
+	    curve = ECC_SECP384R1;
+	} else if (!strncmp(*arg, "P-521", 5)) {
+	    curve = ECC_SECP521R1;
+	} else {
+	    fprintf(stderr, "Invalid Curve Name: %s\n", *arg);
+	    return 1;
+	}
+    }
+
+    ret = gen_key(rng, &ecKey, &rsaKey, type, keySz, exp, curve);
+    if (ret != 0)
+	return ret;
+
+    ret = write_key(&ecKey, &rsaKey, type, keySz, path, pem);
+
+    if (type == EC_KEY_TYPE) {
+	wc_ecc_free(&ecKey);
+    } else {
+	wc_FreeRsaKey(&rsaKey);
+    }
+    return ret;
+}
+
+int main(int argc, char* argv[])
+{
+    int ret;
+    WC_RNG rng;
+    ret = wc_InitRng(&rng);
+    if (ret != 0) {
+	fprintf(stderr, "Init Rng failed: %d\n", ret);
+	return ret;
+    }
+
+    if (!argv[1]) {
```

Comment:
I don't understand this one. This if is blank. Also, the other if statements are just return statements. There's no point in using else after return.

Reply:
I changed the logic a bit